### PR TITLE
T5x loss+perf baselines

### PR DESCRIPTION
**Pull request by maanug-nv**
- print test metrics
- update baseline scripts to support T5x
- add t5x test baselines

### CODE REVIEW

File: .github/workflows/baselines/summarize_metrics.py [hunk 1]
```diff
@@ -24,28 +25,36 @@ def _create_summary(loss, train_time, e2e_time):
...
-        event_file = glob.glob(event_file)[0]
-        loss = read_tb_tag(event_file, loss_summary_name)
-        train_time = read_tb_tag(event_file, train_time_summary_name)
-        e2e_time = read_e2e_time(test_config + ".log")
+        searchpath = os.path.join(args.test_config, "train")
+        if not os.path.exists(searchpath):
+            searchpath = os.path.join(args.test_config, "summaries/train")
+        assert os.path.exists(searchpath), f"Neither {args.test_config}/train nor {args.test_config}/summaries/train dirs exist"
+        event_file = glob.glob(os.path.join(searchpath, "events*"))
+        assert event_file != [], f"{searchpath} did not contain a tensorboard events file"
```
Comment (terrykong) [inline .github/workflows/baselines/summarize_metrics.py:43]: minor nit:

I think `len(event_files) > 0` reads a little better

File: .github/workflows/baselines/T5X_MGMN/1G1N.json
Comment (terrykong) [inline .github/workflows/baselines/T5X_MGMN/1G1N.json:1]: Any luck getting the other benchmark files?

Comment (maanug-nv) [reply .github/workflows/baselines/T5X_MGMN/1G1N.json:1]: Unfortunately not. Launched the following 2 runs as you explained, but still seeing timeout or failure for MGMN jobs (2G1N, 2G2N, 4G1N, ...):
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6313044245
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6313047132

Comment (terrykong) [reply .github/workflows/baselines/T5X_MGMN/1G1N.json:1]: Ahh okay, that info helps. It looks like in coming up with this WAR to retroactively test older containers, I created a race condition since `--tasks-per-node` all tried to write to the same location. Let me propose a fix and you can test again.

**Comment by yhtang** [comment]
@maanug-nv Could you please update the description with an example of the mechanism in effect?

File: .github/workflows/baselines/T5X_MGMN/1G1N.json
Comment (terrykong) [reply .github/workflows/baselines/T5X_MGMN/1G1N.json:1]: Can you try again? This change should hopefully work as a WAR: https://github.com/NVIDIA/JAX-Toolbox/commit/19317eac4f3ffee78b69328cd4a481ef1c30c7fd

**Comment by maanug-nv** [comment]
@yhtang Which description? PR description, or in one of the scripts, or in the README? 
Also do you just want example call of `create_baselines.sh`, or something more? If you can provide more detail, it will be helpful.

File: .github/workflows/baselines/T5X_MGMN/1G1N.json
Comment (maanug-nv) [reply .github/workflows/baselines/T5X_MGMN/1G1N.json:1]: Thanks for the fix, all configs are completing now. I'm using these 3 runs for all the configs now:
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6322101966
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6321648332
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6321558758

**Review by terrykong — DISMISSED** [review]

**Comment by terrykong** [comment]
@yhtang bumping for @maanug-nv 's Q

File: .github/workflows/_test_t5x.yaml [hunk 2]
```diff
@@ -268,6 +268,36 @@ jobs:
...
+            python3 .github/workflows/baselines/summarize_metrics.py $SUBDIR --perf_summary_name "timing/steps_per_second" # create result json in baseline format
+          done
+
+          echo '## T5X MGMN Test Metrics' >> $GITHUB_STEP_SUMMARY
+          for i in *_metrics.json; do
+            echo $i | cut -d'.' -f1
+            echo '```json'
+            jq . $i
+            echo '```'
+          done | tee -a $GITHUB_STEP_SUMMARY
```
Comment (terrykong) [inline .github/workflows/_test_t5x.yaml:298]: Why doesn't t5x also run pytest like the pax tests do?

https://github.com/NVIDIA/JAX-Toolbox/blob/main/.github/workflows/_test_pax.yaml#L187C11-L187C121

Comment (maanug-nv) [reply .github/workflows/_test_t5x.yaml:298]: I was going to add the tests in a separate PR. I guess I might as well add it here now that we have all the baselines.

Comment (terrykong) [reply .github/workflows/_test_t5x.yaml:298]: Yea, let's add it in this PR since you may want to change the baselines if the tests don't pass according to your expectations

**Comment by maanug-nv** [comment]
Added `.github/workflows/baselines/test_t5x_mgmn_metrics.py`, and the `pytest` command. 
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6386266144

File: .github/workflows/_test_t5x.yaml [hunk 3]
```diff
@@ -268,6 +268,37 @@ jobs:
...
+
+          RESULTS_DIR=$PWD pytest --report-log=report.jsonl .github/workflows/baselines/test_t5x_mgmn_metrics.py || true
+
   publish-test:
     needs: [multi-gpu-multi-node, single-process-multi-device]
```
Comment (terrykong) [inline .github/workflows/_test_t5x.yaml:303]: Can we also have metrics be added in `publish-test`: https://github.com/NVIDIA/JAX-Toolbox/blob/main/.github/workflows/_test_pax.yaml#L197

**Comment by maanug-nv** [comment]
I've tuned the relative tolerances a few times, all test cases are passing in this run: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6475410065
Pls review the T5x loss relative tolerances

File: .github/workflows/_test_t5x.yaml [hunk 4]
```diff
@@ -295,18 +333,32 @@ jobs:
         echo "Test statuses:"
         jq -rc 'input_filename,.' $EXIT_STATUSES
 
-        if [[ $FAILED_TESTS -eq 0 ]] && [[ $TOTAL_TESTS -gt 0 ]] || [[ $PASSED_TESTS -eq $TOTAL_TESTS ]]; then
-          echo "STATUS=success" >> $GITHUB_OUTPUT
+        PYTEST_LOG=metrics-test-log/report.jsonl
```
Comment (terrykong) [inline .github/workflows/_test_t5x.yaml:336]: minor nit: I'd rename `PYTEST_*` to `METRIC_*` to convey meaning in this script

File: .github/workflows/_test_t5x.yaml [hunk 5]
```diff
@@ -295,18 +333,32 @@ jobs:
...
+        cnt_type() {
+          cat $PYTEST_LOG | jq '. | select((.["$report_type"] == "TestReport") and (.when == "call") and (.outcome | contains("'${1}'"))) | .outcome' | wc -l
+        }
+        PYTEST_FAILED_TESTS=$(cnt_type failed)
+        PYTEST_PASSED_TESTS=$(cnt_type passed)
+        PYTEST_TOTAL_TESTS=$(all_outcomes | wc -l)
+
+        if ([[ $FAILED_TESTS -eq 0 ]] && [[ $TOTAL_TESTS -gt 0 ]] && \
+            [[ $PYTEST_FAILED_TESTS -eq 0 ]] && [[ $PYTEST_TOTAL_TESTS -gt 0 ]]) || \
+            ([[ $PASSED_TESTS -eq $TOTAL_TESTS ]] && [[ $PYTEST_PASSED_TESTS -eq $PYTEST_TOTAL_TESTS ]]); then
```
Comment (terrykong) [inline .github/workflows/_test_t5x.yaml:349]: Is it possible for the tests in line 349 to `-ne` each other given the previous conditions on 347:348 return true?

Comment (maanug-nv) [reply .github/workflows/_test_t5x.yaml:349]: I think you are saying line 349 is redundant, correct? I can't think of a case where line 349 is true when 347,348 eval to false. I think it's safe to remove

**Comment by terrykong** [comment]
The rtol LGTM. Looks like #292 may have introduced conflicts that you have to resolve in `summarize_metrics.py`, but assuming the CI passes, I think we're good to merge.

**Comment by maanug-nv** [comment]
resolved, if everything passes FFTM

**Comment by terrykong** [comment]
Manual run of t5x MGMN since t5x build is failing due to a known issue: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6492103243

**Review by terrykong — APPROVED** [review]
